### ISSUE TITLE
Symfony 5 incompatibility

### ISSUE THREAD

**Issue by ameotoko** [terminal42/contao-url-rewrite#24]
`contao-url-rewrite` currently cannot be installed with Contao 4.11, which is fine, but there is a problem if I'm trying to upgrade existing 4.10 installation when `contao-url-rewrite` is already installed. `post-update-cmd` fails with `Script @php vendor/bin/contao-setup handling the post-update-cmd event returned with error code 255`, and `contao-console` is broken with the following stack trace:

```
ArgumentCountError {#4049
  #message: "Too few arguments to function Symfony\Component\Config\Definition\Builder\TreeBuilder::__construct(), 0 passed in /Users/andrey/Sites/classic-at-home/vendor/terminal42/contao-url-rewrite/src/DependencyInjection/Configuration.php on line 24 and at least 1 expected"
  #code: 0
  #file: "./vendor/symfony/config/Definition/Builder/TreeBuilder.php"
  #line: 26
  trace: {
    ./vendor/symfony/config/Definition/Builder/TreeBuilder.php:26 { …}
    ./vendor/terminal42/contao-url-rewrite/src/DependencyInjection/Configuration.php:24 { …}
    ./vendor/symfony/config/Definition/Processor.php:50 { …}
    ./vendor/symfony/dependency-injection/Extension/Extension.php:111 { …}
    ./vendor/symfony/http-kernel/DependencyInjection/ConfigurableExtension.php:35 { …}
    ./vendor/symfony/dependency-injection/Compiler/MergeExtensionConfigurationPass.php:76 { …}
    ./vendor/symfony/http-kernel/DependencyInjection/MergeExtensionConfigurationPass.php:39 { …}
    ./vendor/symfony/dependency-injection/Compiler/Compiler.php:91 { …}
    ./vendor/symfony/dependency-injection/ContainerBuilder.php:736 { …}
    ./vendor/symfony/http-kernel/Kernel.php:541 { …}
    ./vendor/contao/manager-bundle/src/HttpKernel/ContaoKernel.php:311 { …}
    ./vendor/symfony/http-kernel/Kernel.php:780 { …}
    ./vendor/symfony/http-kernel/Kernel.php:121 { …}
    ./vendor/symfony/framework-bundle/Console/Application.php:168 { …}
    ./vendor/symfony/framework-bundle/Console/Application.php:74 { …}
    ./vendor/symfony/console/Application.php:166 { …}
    ./vendor/contao/manager-bundle/bin/contao-console:37 { …}
  }
}
```

On a project level, this can be solved by locking Symfony at `4.4.*`.

See also https://github.com/contao/contao/issues/2844#issuecomment-790986503

**Comment by Flo-JB** [comment]
Even if you pass the TreeBuilder-Error you will get an error later on creating new redirect rules because the router option "generator_cache_class" isn't available anymore in Symfony 5

[](https://github.com/symfony/symfony/blob/5.x/UPGRADE-5.0.md)

Maybe additional changes are needed

**Comment by adera-web** [comment]
Yes, I confirm: terminal42/contao-url-rewrite prevents contao upgrade to 4.11
Install-Tool fails and I get in SSH:

Fatal error: Uncaught ArgumentCountError: Too few arguments to function Symfony\Component\Config\Definition\Builder\TreeBuilder::__construct(), 0 passed in /www/htdocs/domain.tld/vendor/terminal42/contao-url-rewrite/src/DependencyInjection/Configuration.php on line 24 and at least 1 expected in /www/htdocs/domain.tld/vendor/symfony/config/Definition/Builder/TreeBuilder.php:26
Stack trace:
#0 /www/htdocs/domain.tld/vendor/terminal42/contao-url-rewrite/src/DependencyInjection/Configuration.php(24): Symfony\Component\Config\Definition\Builder\TreeBuilder->__construct()
#1 /www/htdocs/domain.tld/vendor/symfony/config/Definition/Processor.php(50): Terminal42\UrlRewriteBundle\DependencyInjection\Configuration->getConfigTreeBuilder()
#2 /www/htdocs/domain.tld/vendor/symfony/dependency-injection/Extension/Extension.php(111): Symfony\Component\Config\Definition\Processor->processConfiguration(Object(Terminal42\UrlRewriteBundle\DependencyInjection\Configuration), Arra in /www/htdocs/domain.tld/vendor/symfony/config/Definition/Builder/TreeBuilder.php on line 26
PHP Fatal error:  Uncaught ArgumentCountError: Too few arguments to function Symfony\Component\Config\Definition\Builder\TreeBuilder::__construct(), 0 passed in /www/htdocs/domain.tld/vendor/terminal42/contao-url-rewrite/src/DependencyInjection/Configuration.php on line 24 and at least 1 expected in /www/htdocs/domain.tld/vendor/symfony/config/Definition/Builder/TreeBuilder.php:26
Stack trace:
#0 /www/htdocs/domain.tld/vendor/terminal42/contao-url-rewrite/src/DependencyInjection/Configuration.php(24): Symfony\Component\Config\Definition\Builder\TreeBuilder->__construct()
#1 /www/htdocs/domain.tld/vendor/symfony/config/Definition/Processor.php(50): Terminal42\UrlRewriteBundle\DependencyInjection\Configuration->getConfigTreeBuilder()
#2 /www/htdocs/domain.tld/vendor/symfony/dependency-injection/Extension/Extension.php(111): Symfony\Component\Config\Definition\Processor->processConfiguration(Object(Terminal42\UrlRewriteBundle\DependencyInjection\Configuration), Arra in /www/htdocs/domain.tld/vendor/symfony/config/Definition/Builder/TreeBuilder.php on line 26

**Comment by fritzmg** [comment]
> Yes, I confirm: terminal42/contao-url-rewrite prevents contao upgrade to 4.11

The update itself was not prevented. The error happens after updating, because Symfony gets updated to version 5 as well with Contao 4.11. As already mentioned in the initial post you can work around the error for now by forcing the installation of Symfony 4.

**Comment by rabauss** [comment]
Should be fixed with #25
